pkg/api/v1/router_server_components.go

```diff
@@ -339,39 +339,56 @@ func (r *Router) serverComponentUpdate(c *gin.Context) {
 				return
 			}
 
+			dbAttributes.ServerComponentID = null.StringFrom(dbSrvComponent.ID)
+
+			// upsert component attribute data
+
+			//
+			// This update, insert could have been swapped with the sqlboil Upsert() method
+			// although since the attributes table contains a partial index - "WHERE server_component_id is not null"
+			// and the sqlboil Upsert() method has no way of specifying query mods,
+			// the current Upsert() method does not work for this case (the matching row for update is not found in the row scan).
+			//
+			// https://github.com/volatiletech/sqlboiler/issues/856
+			//
+
+			// update attribute when an attribute matching server_component_id, namespace was found
 			match, err := models.Attributes(
 				qm.Where("server_component_id=?", dbSrvComponent.ID),
```

Comment:
Should this be `dbAttributes.ServerComponentID` instead of `dbSrvComponent.ID`?

Reply:
`dbAttributes.ServerComponentID` is being assigned the value from `dbSrvComponent.ID` in https://github.com/metal-toolbox/hollow-serverservice/blob/d57f459af96e4f30252f1d511ce71d6c9a8e6951/pkg/api/v1/router_server_components.go#L342 so they are they same here

Reply:
Yeah, I know. It just seemed weird that you assigned this on the line above and then didn't use it

Reply:
oh right, yeah its being assigned to `dbAttributes` for the insert, which is further below.